test/parallel/test-domain-promise.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const domain = require('domain');
+const vm = require('vm');
+
+const d = domain.create();
+
```

Comment:
@matthewloring thanks for letting us know, looks pretty sensible to me :) Also seems pretty doable to integrate it with the work in this PR